src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthModule.TransactionExecutor.cs

```diff
@@ -0,0 +1,180 @@
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System;
+using System.Threading;
+using Nethermind.Blockchain.Find;
+using Nethermind.Core;
+using Nethermind.Core.Eip2930;
+using Nethermind.Core.Extensions;
+using Nethermind.Evm;
+using Nethermind.Facade;
+using Nethermind.Int256;
+using Nethermind.JsonRpc.Data;
+using Nethermind.Specs.Forks;
+
+namespace Nethermind.JsonRpc.Modules.Eth
+{
+    public partial class EthModule
+    {
+        private abstract class TransactionExecutor<T>
+        {
+            protected readonly IBlockchainBridge _blockchainBridge;
+            private readonly IBlockFinder _blockFinder;
+            private readonly TimeSpan _cancellationTokenTimeout;
+            private readonly IJsonRpcConfig _rpcConfig;
+
+            protected TransactionExecutor(IBlockchainBridge blockchainBridge, IBlockFinder blockFinder, TimeSpan cancellationTokenTimeout, IJsonRpcConfig rpcConfig)
+            {
+                _blockchainBridge = blockchainBridge;
+                _blockFinder = blockFinder;
+                _cancellationTokenTimeout = cancellationTokenTimeout;
+                _rpcConfig = rpcConfig;
+            }
+            
+            public ResultWrapper<T> ExecuteTx(
+                TransactionForRpc transactionCall, 
+                BlockParameter? blockParameter)
+            {
+                SearchResult<BlockHeader> searchResult = _blockFinder.SearchForHeader(blockParameter);
+                if (searchResult.IsError)
+                {
+                    return ResultWrapper<T>.Fail(searchResult);
+                }
+
+                BlockHeader header = searchResult.Object;
+                if (!HasStateForBlock(_blockchainBridge, header))
+                {
+                    return ResultWrapper<T>.Fail($"No state available for block {header.Hash}",
+                        ErrorCodes.ResourceUnavailable);
+                }
+
+                FixCallTx(transactionCall);
+
+                using CancellationTokenSource cancellationTokenSource = new(_cancellationTokenTimeout);
+                Transaction tx = transactionCall.ToTransaction(_blockchainBridge.GetChainId());
+                BlockchainBridge.CallOutput result = ExecuteTx(header, tx, cancellationTokenSource.Token);
+
+                if (result.Error == null)
+                {
+                    return ResultWrapper<T>.Success(GetResult(tx, result));
+                }
+
+                return result.InputError
+                    ? GetInputError(result)
+                    : GetExecutionError(tx, result);
+            }
+
+            private ResultWrapper<T> GetInputError(BlockchainBridge.CallOutput result) => 
+                ResultWrapper<T>.Fail(result.Error, ErrorCodes.InvalidInput);
+
+            protected abstract BlockchainBridge.CallOutput ExecuteTx(BlockHeader header, Transaction tx, CancellationToken token);
```

Comment:
remove these 3 small asbtract calls and embed them inside the ExecuteTx abstract call to simplify the code